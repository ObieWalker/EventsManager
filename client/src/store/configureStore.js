import { createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { createLogger } from 'redux-logger';
import rootReducer from '../reducers/rootReducer';

const middleware = applyMiddleware(thunkMiddleware, createLogger({ collapsed: true }));

const Store = createStore(
    rootReducer,
    applyMiddleware(middleware)
);

export default Store;