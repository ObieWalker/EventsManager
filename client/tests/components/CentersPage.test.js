import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import connect, { CentersPage }
  from '../../src/components/UserDashboard/CentersPage';
import { centers } from '../__mocks__/centersData';

let mountedComponent;
let props;
const mockStore = configureStore();
// const createCenter = jest.fn(() => Promise.resolve({}));

let wrapper;
const allCenters = {
  centers,
  moreCenters: false
};


const getAllCenters = jest.fn(() => Promise.resolve({}));
const createSuccess = '';
const createError = '';
const Center = { name: 'random' };
let moreCenters, dumbObject = {};
const centerEvents = jest.fn();
const fetchedCenterEvents = [
  dumbObject = {
    Center: {
      name: 'random'
    }

  }
];
const clearCenterEvents = jest.fn();
const addNewEvent = jest.fn(() => Promise.resolve({}));
const getComponent = () => {
  if (!mountedComponent) {
    props = {
      allCenters,
      getAllCenters,
      createSuccess,
      createError,
      addNewEvent,
      Center,
      moreCenters,
      centerEvents,
      fetchedCenterEvents,
      clearCenterEvents
    };
    mountedComponent = shallow(<CentersPage {...props} />);
  }
  return mountedComponent;
};

describe('Center Page Component', () => {
  beforeEach(() => {});
  wrapper = getComponent();

  it('component successfully rendered', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a method that handles change to each input element', () => {
    expect(wrapper.instance().loadMoreContent).toBeDefined();
  });
  it('should have a method that handles change to each input element', () => {
    expect(wrapper.instance().getMoreCenters).toBeDefined();
  });
  it('should have a method that handles handleDateChange', () => {
    expect(wrapper.instance().handleClose).toBeDefined();
  });
  it('should have a method that handles handleDateChange', () => {
    expect(wrapper.instance().handleShowModal).toBeDefined();
  });
  it('should have a method that handles handleDateChange', () => {
    expect(wrapper.instance().changeSearchState).toBeDefined();
  });
  it('should have a method that handles handleDateChange', () => {
    expect(wrapper.instance().getCenterEvents).toBeDefined();
  });
  it('should have a method that handles handleDateChange', () => {
    expect(wrapper.instance().revertPageState).toBeDefined();
  });
  it('should have a method that handles handleDateChange', () => {
    expect(wrapper.instance().resetPage).toBeDefined();
  });

  it('testing clear function', () => {
    wrapper = getComponent();
    wrapper.instance().resetPage();
    expect(wrapper.state().pageNo).toEqual(1);
    expect(wrapper.state().filter).toEqual('');
    expect(wrapper.state().capacity).toEqual('');
    expect(wrapper.state().facility).toEqual('');
  });
  it('testing get center events function', () => {
    wrapper = getComponent();
    const center = {
      id: 5
    };
    wrapper.instance().getCenterEvents(center);
    expect(wrapper.state().showCenterEvents).toEqual(true);
  });
  it('testing revert page state function', () => {
    wrapper = getComponent();
    const e = { preventDefault: () => undefined };
    e.preventDefault = jest.fn();
    wrapper.instance().revertPageState(e);
    expect(e.preventDefault).toHaveBeenCalled();
    expect(wrapper.state().showCenterEvents).toEqual(false);
  });
  it('testing handleShowModal function', () => {
    wrapper = getComponent();
    wrapper.instance().handleShowModal();
    expect(wrapper.state().show).toEqual(true);
  });
});

describe('Connected Component', () => {
  describe('Connected mountedLogin', () => {
    it('component successfully rendered', () => {
      const store = mockStore({
        allCenters,
        getAllCenters,
        createSuccess,
        createError,
        addNewEvent,
        Center,
        moreCenters,
        centerEvents,
        fetchedCenterEvents,
        clearCenterEvents
      });
      wrapper = shallow(<connect store={store}
        allCenters={allCenters}
        getAllCenters={getAllCenters}
        createSuccess={createSuccess}
        createError={createError}
        addNewEvent={addNewEvent}
        Center={Center}
        moreCenters={moreCenters}
        centerEvents={centerEvents}
        fetchedCenterEvents={fetchedCenterEvents}
        clearCenterEvents={clearCenterEvents}
      />);
      expect(wrapper.length).toBe(1);
    });
  });
});
