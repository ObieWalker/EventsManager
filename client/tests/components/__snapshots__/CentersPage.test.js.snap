// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Center Page Component component successfully rendered 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CentersPage
    Center={
      Object {
        "name": "random",
      }
    }
    addNewEvent={[MockFunction]}
    allCenters={
      Object {
        "centers": Object {
          "center1": Object {
            "address": "123, Address I Like",
            "capacity": 500,
            "city": "Lagos",
            "createdAt": "2018-05-25T18:04:41.552Z",
            "facility": "Parking, Pool and 24hr security",
            "id": 15,
            "image": null,
            "name": "The Best Center",
            "updatedAt": "2018-05-25T18:04:41.552Z",
            "userId": 1,
          },
          "center2": Object {
            "address": "Another random address",
            "capacity": 2000,
            "city": "Lagos",
            "createdAt": "2018-07-25T18:04:41.552Z",
            "facility": "Facilities Jabarata",
            "id": 17,
            "image": null,
            "name": "Center two",
            "updatedAt": "2018-07-25T18:04:41.552Z",
            "userId": 2,
          },
          "center3": Object {
            "address": "7, Ikoyi Road, Lekki",
            "capacity": 3000,
            "city": "Lagos",
            "createdAt": "2018-04-25T18:04:41.552Z",
            "facility": "Wonderful mezanine",
            "id": 15,
            "image": null,
            "name": "Center three",
            "updatedAt": "2018-04-25T18:04:41.552Z",
            "userId": 2,
          },
        },
        "moreCenters": false,
      }
    }
    centerEvents={[MockFunction]}
    clearCenterEvents={[MockFunction]}
    createError=""
    createSuccess=""
    fetchedCenterEvents={
      Array [
        Object {
          "Center": Object {
            "name": "random",
          },
        },
      ]
    }
    getAllCenters={
      [MockFunction] {
        "calls": Array [
          Array [
            1,
            6,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": Promise {},
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <h3>
            Available Centers.
          </h3>
          <div
            className="center"
          >
            <Connect(Search)
              changeSearchState={[Function]}
              resetPage={[Function]}
            />
             
            <br />
             
            <br />
          </div>
          <div>
             
            There are no registered centers.
            <ScrollUp
              duration={250}
              easing="easeOutCubic"
              showUnder={100}
              style={
                Object {
                  "bottom": 50,
                  "cursor": "pointer",
                  "position": "fixed",
                  "right": 30,
                  "transitionDelay": "0s",
                  "transitionDuration": "0.2s",
                  "transitionTimingFunction": "linear",
                }
              }
              topPosition={0}
            >
              <button
                className="btn btn-floating btn-rounded waves-effect"
                type="button"
              >
                TOP
              </button>
            </ScrollUp>
            <br />
            <br />
          </div>
        </div>,
        <Modal
          animation={true}
          autoFocus={true}
          backdrop={true}
          bsClass="modal"
          bsSize="large"
          className="modal-display"
          dialogComponentClass={[Function]}
          enforceFocus={true}
          keyboard={true}
          manager={
            ModalManager {
              "add": [Function],
              "containers": Array [],
              "data": Array [],
              "handleContainerOverflow": true,
              "hideSiblingNodes": true,
              "isTopModal": [Function],
              "modals": Array [],
              "remove": [Function],
            }
          }
          onHide={[Function]}
          renderBackdrop={[Function]}
          restoreFocus={true}
          show={false}
        >
          <Connect(BookCenter)
            center=""
            onHide={[Function]}
          />
        </Modal>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              Available Centers.
            </h3>,
            <div
              className="center"
            >
              <Connect(Search)
                changeSearchState={[Function]}
                resetPage={[Function]}
              />
               
              <br />
               
              <br />
            </div>,
            <div>
               
              There are no registered centers.
              <ScrollUp
                duration={250}
                easing="easeOutCubic"
                showUnder={100}
                style={
                  Object {
                    "bottom": 50,
                    "cursor": "pointer",
                    "position": "fixed",
                    "right": 30,
                    "transitionDelay": "0s",
                    "transitionDuration": "0.2s",
                    "transitionTimingFunction": "linear",
                  }
                }
                topPosition={0}
              >
                <button
                  className="btn btn-floating btn-rounded waves-effect"
                  type="button"
                >
                  TOP
                </button>
              </ScrollUp>
              <br />
              <br />
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Available Centers.",
            },
            "ref": null,
            "rendered": "Available Centers.",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Connect(Search)
                  changeSearchState={[Function]}
                  resetPage={[Function]}
                />,
                " ",
                <br />,
                " ",
                <br />,
              ],
              "className": "center",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "changeSearchState": [Function],
                  "resetPage": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "There are no registered centers.",
                false,
                <ScrollUp
                  duration={250}
                  easing="easeOutCubic"
                  showUnder={100}
                  style={
                    Object {
                      "bottom": 50,
                      "cursor": "pointer",
                      "position": "fixed",
                      "right": 30,
                      "transitionDelay": "0s",
                      "transitionDuration": "0.2s",
                      "transitionTimingFunction": "linear",
                    }
                  }
                  topPosition={0}
                >
                  <button
                    className="btn btn-floating btn-rounded waves-effect"
                    type="button"
                  >
                    TOP
                  </button>
                </ScrollUp>,
                undefined,
                <br />,
                <br />,
              ],
            },
            "ref": null,
            "rendered": Array [
              " ",
              "There are no registered centers.",
              false,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <button
                    className="btn btn-floating btn-rounded waves-effect"
                    type="button"
                  >
                    TOP
                  </button>,
                  "duration": 250,
                  "easing": "easeOutCubic",
                  "showUnder": 100,
                  "style": Object {
                    "bottom": 50,
                    "cursor": "pointer",
                    "position": "fixed",
                    "right": 30,
                    "transitionDelay": "0s",
                    "transitionDuration": "0.2s",
                    "transitionTimingFunction": "linear",
                  },
                  "topPosition": 0,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "TOP",
                    "className": "btn btn-floating btn-rounded waves-effect",
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": "TOP",
                  "type": "button",
                },
                "type": [Function],
              },
              undefined,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "animation": true,
          "autoFocus": true,
          "backdrop": true,
          "bsClass": "modal",
          "bsSize": "large",
          "children": <Connect(BookCenter)
            center=""
            onHide={[Function]}
          />,
          "className": "modal-display",
          "dialogComponentClass": [Function],
          "enforceFocus": true,
          "keyboard": true,
          "manager": ModalManager {
            "add": [Function],
            "containers": Array [],
            "data": Array [],
            "handleContainerOverflow": true,
            "hideSiblingNodes": true,
            "isTopModal": [Function],
            "modals": Array [],
            "remove": [Function],
          },
          "onHide": [Function],
          "renderBackdrop": [Function],
          "restoreFocus": true,
          "show": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "center": "",
            "onHide": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <h3>
              Available Centers.
            </h3>
            <div
              className="center"
            >
              <Connect(Search)
                changeSearchState={[Function]}
                resetPage={[Function]}
              />
               
              <br />
               
              <br />
            </div>
            <div>
               
              There are no registered centers.
              <ScrollUp
                duration={250}
                easing="easeOutCubic"
                showUnder={100}
                style={
                  Object {
                    "bottom": 50,
                    "cursor": "pointer",
                    "position": "fixed",
                    "right": 30,
                    "transitionDelay": "0s",
                    "transitionDuration": "0.2s",
                    "transitionTimingFunction": "linear",
                  }
                }
                topPosition={0}
              >
                <button
                  className="btn btn-floating btn-rounded waves-effect"
                  type="button"
                >
                  TOP
                </button>
              </ScrollUp>
              <br />
              <br />
            </div>
          </div>,
          <Modal
            animation={true}
            autoFocus={true}
            backdrop={true}
            bsClass="modal"
            bsSize="large"
            className="modal-display"
            dialogComponentClass={[Function]}
            enforceFocus={true}
            keyboard={true}
            manager={
              ModalManager {
                "add": [Function],
                "containers": Array [],
                "data": Array [],
                "handleContainerOverflow": true,
                "hideSiblingNodes": true,
                "isTopModal": [Function],
                "modals": Array [],
                "remove": [Function],
              }
            }
            onHide={[Function]}
            renderBackdrop={[Function]}
            restoreFocus={true}
            show={false}
          >
            <Connect(BookCenter)
              center=""
              onHide={[Function]}
            />
          </Modal>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Available Centers.
              </h3>,
              <div
                className="center"
              >
                <Connect(Search)
                  changeSearchState={[Function]}
                  resetPage={[Function]}
                />
                 
                <br />
                 
                <br />
              </div>,
              <div>
                 
                There are no registered centers.
                <ScrollUp
                  duration={250}
                  easing="easeOutCubic"
                  showUnder={100}
                  style={
                    Object {
                      "bottom": 50,
                      "cursor": "pointer",
                      "position": "fixed",
                      "right": 30,
                      "transitionDelay": "0s",
                      "transitionDuration": "0.2s",
                      "transitionTimingFunction": "linear",
                    }
                  }
                  topPosition={0}
                >
                  <button
                    className="btn btn-floating btn-rounded waves-effect"
                    type="button"
                  >
                    TOP
                  </button>
                </ScrollUp>
                <br />
                <br />
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Available Centers.",
              },
              "ref": null,
              "rendered": "Available Centers.",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Connect(Search)
                    changeSearchState={[Function]}
                    resetPage={[Function]}
                  />,
                  " ",
                  <br />,
                  " ",
                  <br />,
                ],
                "className": "center",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "changeSearchState": [Function],
                    "resetPage": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "There are no registered centers.",
                  false,
                  <ScrollUp
                    duration={250}
                    easing="easeOutCubic"
                    showUnder={100}
                    style={
                      Object {
                        "bottom": 50,
                        "cursor": "pointer",
                        "position": "fixed",
                        "right": 30,
                        "transitionDelay": "0s",
                        "transitionDuration": "0.2s",
                        "transitionTimingFunction": "linear",
                      }
                    }
                    topPosition={0}
                  >
                    <button
                      className="btn btn-floating btn-rounded waves-effect"
                      type="button"
                    >
                      TOP
                    </button>
                  </ScrollUp>,
                  undefined,
                  <br />,
                  <br />,
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                "There are no registered centers.",
                false,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <button
                      className="btn btn-floating btn-rounded waves-effect"
                      type="button"
                    >
                      TOP
                    </button>,
                    "duration": 250,
                    "easing": "easeOutCubic",
                    "showUnder": 100,
                    "style": Object {
                      "bottom": 50,
                      "cursor": "pointer",
                      "position": "fixed",
                      "right": 30,
                      "transitionDelay": "0s",
                      "transitionDuration": "0.2s",
                      "transitionTimingFunction": "linear",
                    },
                    "topPosition": 0,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "TOP",
                      "className": "btn btn-floating btn-rounded waves-effect",
                      "type": "button",
                    },
                    "ref": null,
                    "rendered": "TOP",
                    "type": "button",
                  },
                  "type": [Function],
                },
                undefined,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "animation": true,
            "autoFocus": true,
            "backdrop": true,
            "bsClass": "modal",
            "bsSize": "large",
            "children": <Connect(BookCenter)
              center=""
              onHide={[Function]}
            />,
            "className": "modal-display",
            "dialogComponentClass": [Function],
            "enforceFocus": true,
            "keyboard": true,
            "manager": ModalManager {
              "add": [Function],
              "containers": Array [],
              "data": Array [],
              "handleContainerOverflow": true,
              "hideSiblingNodes": true,
              "isTopModal": [Function],
              "modals": Array [],
              "remove": [Function],
            },
            "onHide": [Function],
            "renderBackdrop": [Function],
            "restoreFocus": true,
            "show": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "center": "",
              "onHide": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
